File: Problems/0398-random-pick-index.py
import random
from collections import defaultdict

class Solution(object):

    def __init__(self, nums):
        """
        :type nums: List[int]
        """
        self.pos = defaultdict(list)
        for i, x in enumerate(nums):
            self.pos[x].append(i)

    def pick(self, target):
        """
        :type target: int
        :rtype: int
        """
        return random.choice(self.pos[target])

File: Problems/0398-random-pick-index.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
    unordered_map<int, vector<int>> pos;

public:
    Solution(vector<int>& nums) {
        for (int i = 0; i < nums.size(); i++)
            pos[nums[i]].push_back(i);
    }

    int pick(int target) {
        vector<int>& v = pos[target];
        return v[rand() % v.size()];
    }
};
